appeals/web/src/server/app/components/file-downloader.component.js

```diff
@@ -60,4 +60,61 @@ const getDocumentDownload = async ({ apiClient, params, session }, response) =>
 	return response.status(200);
 };
 
-export default getDocumentDownload;
+/**
+ * Download one document or redirects to its url if preview is active
+ *
+ * @param {{apiClient: import('got').Got, params: {caseId: number, guid: string, preview?: string, version: string}, session: SessionWithAuth}} request
+ * @param {Response} response
+ * @returns {Promise<Response>}
+ */
+export const getDocumentDownloadByVersion = async ({ apiClient, params, session }, response) => {
+	const { guid: fileGuid, preview, caseId, version } = params;
+
+	const filesInfo = await apiClient.get(`appeals/${caseId}/documents/${fileGuid}/versions`).json();
+	const fileToDownloadInfo = filesInfo.documentVersion.find(
+		(/** @type {{ version: { toString: () => string; }; }} */ fileInfo) =>
+			fileInfo.version.toString() === version
+	);
+	const { blobStorageContainer, blobStoragePath } = fileToDownloadInfo;
+
+	if (!blobStorageContainer || !blobStoragePath) {
+		throw new Error('Blob storage container or Blob storage path not found');
+	}
+
+	let blobStorageClient = undefined;
+	let accessToken = undefined;
+	if (config.useBlobEmulator !== true) {
+		accessToken = await getActiveDirectoryAccessToken(session);
```

Comment:
If `accessToken` isn't used outside this block you could declare it as `const` inside the block

Reply:
Done